File: lintcode/integer-to-roman.py
#!/usr/bin/env python
# encoding: utf-8

# Question: http://www.lintcode.com/en/problem/integer-to-roman/

class Solution:
    # @param {int} n The integer
    # @return {string} Roman representation
    def intToRoman(self, n):
        # Write your code here
        symbol = ["M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"]
        value = [1000,900,500,400,100,90,50,40,10,9,5,4,1]
        ans = []
        i = 0
        while n != 0:
            if n >= value[i]:
                n -= value[i]
                ans.append(symbol[i])
            else:
                i += 1
        return ''.join(ans)

File: leetcode/integer-to-roman.cc
// Question: https://leetcode.com/problems/integer-to-roman/

#include "headers.h"

class Solution {
public:
    string intToRoman(int num) {
        vector<string> symbol{"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        vector<int> value{1000,900,500,400,100,90,50,40,10,9,5,4,1};
        int i = 0;
        string ans("");
        while (num != 0) {
            if (num >= value[i]) {
                num -= value[i];
                ans += symbol[i];
            } else {
                i += 1;
            }
        }
        return ans;
    }
};
